js/src/Ice/ConnectionI.js

```diff
@@ -1042,11 +1015,14 @@ export class ConnectionI {
         os.writeInt(Protocol.headerSize); // Message size.
 
         scheduleCloseTimeout(this);
-        this.sendMessage(OutgoingMessage.createForStream(os));
+
+        if ((this.sendMessage(OutgoingMessage.createForStream(os)) & AsyncStatus.Sent) > 0) {
+            this.setState(StateClosingPending);
```

Comment:
After that, we wait for the peer to abort the transport connection, which transitions our connection to StateClosed.